src/main/java/edu/hm/hafner/analysis/parser/TrivyParser.java

```diff
@@ -0,0 +1,100 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.text.MessageFormat;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.IssueParser;
+import edu.hm.hafner.analysis.ParsingCanceledException;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * <p>
+ * Parser for reports of aquasec trivy container vulnerability scanner.
+ * </p>
+ * <p>
+ * <strong>Usage: </strong>trivy image -f json -o results.json golang:1.12-alpine
+ * </p>
+ *
+ * @author Thomas Fürer - tfuerer.javanet@gmail.com
+ */
+public class TrivyParser extends IssueParser {
+    private static final String TRIVY_VULNERABILITY_LEVEL_TAG_CRITICAL = "critcal";
+    private static final String TRIVY_VULNERABILITY_LEVEL_TAG_HIGH = "high";
+    private static final String TRIVY_VULNERABILITY_LEVEL_TAG_MEDIUM = "medium";
+    private static final String TRIVY_VULNERABILITY_LEVEL_TAG_LOW = "low";
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public Report parse(final ReaderFactory readerFactory) throws ParsingException, ParsingCanceledException {
+        final Report report = new Report();
+
+        try (Reader reader = readerFactory.create()) {
+            final JSONArray jsonReport = (JSONArray)new JSONTokener(reader).nextValue();
+
+            for (int i = 0; i < jsonReport.length(); i++) {
+                final JSONObject component = (JSONObject)jsonReport.get(i);
+                if (!component.isNull("Vulnerabilities")) {
+                    final JSONArray vulnatbilites = component.getJSONArray("Vulnerabilities");
+                    for (Object vulnatbility : vulnatbilites) {
+                        report.add(convertToIssue((JSONObject)vulnatbility));
+                    }
+                }
+            }
+        }
+        catch (IOException e) {
```

Comment:
I think this should be better:
```
catch (IOException | JSONException | ClassCastException e) {
```

Reply:
normally I want to reduce the amount of code, especially when it is not essentially. In this case it is not important to know all these exceptions...
BTW: I'm also not a friend of butting RuntimeExceptions to the throws clause. Standard java rules in SonarLint will claim these approach.